src/app/components/product-table/product-table.harness.ts

```diff
@@ -1,13 +1,30 @@
 import { ComponentHarness } from '@angular/cdk/testing';
 import { MatProgressSpinnerHarness } from '@angular/material/progress-spinner/testing';
+import { MatIconHarness } from '@angular/material/icon/testing';
+import { MatToolbarHarness } from '@angular/material/toolbar/testing';
+import { MatTooltipHarness } from '@angular/material/tooltip/testing';
 
 export class ProductTableHarness extends ComponentHarness {
   static hostSelector = 'tk-product-table';
 
   getLoadingSpinner = this.locatorForOptional(MatProgressSpinnerHarness);
 
+  getTooltips = this.locatorForAll(MatTooltipHarness);
+
   async isLoading(): Promise<boolean> {
     const spinner = await this.getLoadingSpinner();
     return Boolean(spinner);
   }
+
+  async getTooltipText(id: number): Promise<string | null> {
+    const tooltips = await this.getTooltips();
+    const targetTooltip = tooltips[id];
+    if (targetTooltip) {
```

Comment:
why you haven't used getTooltipText ?
https://material.angular.io/components/tooltip/api#MatTooltipHarness

Reply:
I use it in this function, it's just that to get the text you need to have this tooltip be shown, otherwise, it returns nothing
I just made it one function that would open the one that I need and retrieve the text